src/marketplace/ListingsView.js

```diff
@@ -32,117 +38,171 @@ class ListingsView extends React.Component<Props, State> {
 		super(props);
 		this.state={
 			userId: '',
+			chassisForSale: [],
+			weaponsForSale: [],
+			scannersForSale: [],
+			scannerAddonsForSale: [],
+			jammersForSale: [],
+			treadsForSale: [],
 			itemsForSale: [],
-			tanksForSale: null,
+			tanksForSale: [],
+			tanks: [],
+			cards: [<div key={0}><h5>Loading Sales...</h5></div>]
 		}
 	}
 
-	// Once mounted, get the user's ID.
-	componentDidMount() {
-		this.getUserId();
+	// Once mounted, get the user's ID and set the sales.
+	componentDidMount(): void {
+		getUserAPICall(user => {
+			this.setState({userId: user.userId});
+			this.getSales();
```

Comment:
getSales looks at this.state.userId, so shouldn't this be:
`setState({userId: user.userId}, this.getSales);`?

Reply:
Good catch.